src/OrbitHttp/HttpDownloadManager.cpp

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "OrbitHttp/HttpDownloadManager.h"
+
+namespace orbit_http {
+using orbit_base::CanceledOr;
+using orbit_base::Future;
+using orbit_base::Promise;
+using orbit_base::StopToken;
+using orbit_http_internal::HttpDownloadOperation;
+
+HttpDownloadManager::HttpDownloadManager()
+    : QObject(nullptr), manager_(std::make_unique<QNetworkAccessManager>()) {}
+
+HttpDownloadManager::~HttpDownloadManager() {
+  waiting_download_operations_ = std::queue<HttpDownloadOperationMetadata>();
```

Comment:
Abort the current download operation will invoke the finish handler. If the queue is not empty at this time, it will invoke the DoDownload method for the front of the queue.

Reply:
Obsolete after changing to download all.